mockall_derive/src/mock_trait.rs

```diff
@@ -176,3 +174,35 @@ impl MockTrait {
         )
     }
 }
+
+fn join_path_segments(path: &Path, sep: &str) -> String {
+    let mut output = String::new();
+    for segment in &path.segments {
+        if write!(
```

Comment:
Too bad there isn't something like a `Punctuated::join` method to help with this part.